src/Result.ts

```diff
@@ -17,13 +17,13 @@ class ResultClass<Ok, Error> {
       return this as unknown as Result<ReturnValue, Error>;
     }
   }
-  flatMap<ReturnValue>(
-    f: (value: Ok) => Result<ReturnValue, Error>,
-  ): Result<ReturnValue, Error> {
+  flatMap<ReturnValue, ResultError = Error>(
+    f: (value: Ok) => Result<ReturnValue, ResultError | Error>,
+  ): Result<ReturnValue, ResultError | Error> {
```

Comment:
Shouldn't it be `ResultError | Error` instead? If I precise the Error type on call, I don't think it's great to have `| Error`